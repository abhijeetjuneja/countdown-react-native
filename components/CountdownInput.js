import React, { Component } from 'react';
import { StyleSheet, Text, View, TextInput, Button, TouchableOpacity } from 'react-native';

export default class CountdownInput extends Component {
    
    constructor(props) {
        super(props)
        this.state = {
            key: ''
        }
    }


    render() {
        let { countdown, start } = this.props
        return (
            <View style={styles.container}>
                <Text style={styles.inputLabel}>Countdown</Text><br /><br />
                    <Text>
                    <TextInput
                        style={styles.textInput}
                        placeholder="Enter minutes"
                        onChange={(e) => {
                            const re = /^[0-9\b]+$/;
                            if (e.target.value === '' || re.test(e.target.value)) {
                                this.props.setStates('countdown',e.target.value)}
                            }
                        }
                        value={countdown}
                        disabled={start}
                    />
                    {start ? 
                    <TouchableOpacity
                        style={countdown <= 0 ? styles.disabledButton : styles.buttonStyle2}
                        onPress={() => {this.props.setStates('start',false), this.props.setStates('countdown',0)} }
                    ><Text style={countdown <= 0 ? styles.disabledButtonText : styles.buttonText}>STOP</Text></TouchableOpacity> : 
                    <TouchableOpacity
                        style={countdown <= 0 ? styles.disabledButton : styles.buttonStyle1}
                        onPress={() => this.props.setStates('start',true)}
                        disabled={countdown <= 0}
                    ><Text style={countdown <= 0 ? styles.disabledButtonText : styles.buttonText}>START</Text></TouchableOpacity>}</Text>
            </View>
        )
    }
}


const styles = StyleSheet.create({
    container:{
        marginTop:50,
        alignItems: 'center',
        justifyContent: 'center',
    },
    inputLabel: {
        color:'white',
        backgroundColor:'#0066FF',
        fontSize:'30px',
        width:'100%',
        fontWeight:'700',
        paddingLeft:50,
        paddingRight:50,
        paddingTop:15,
        paddingBottom:15,
        border: '2px solid #0066FF',
        borderRadius:10
    },
    textInput: {
        height: 50,
        width: '150px',
        fontSize:'16px',
        paddingLeft:12,
        border: '1px solid black',
        borderRadius:10,
        alignSelf:'center'
    },
    buttonText: {
        color:'white',
        fontSize:'16px',
        fontWeight:500
    },
    disabledButtonText: {
        color:'grey',
        fontSize:'16px',
        fontWeight:'500',
    },
    disabledButton:{
        height: 50,
        marginTop:10,
        paddingTop:15,
        paddingBottom:15,
        paddingLeft:30,
        paddingRight:30,
        marginLeft:10,
        backgroundColor:'#C0C0C0',
        borderRadius:10,
        borderWidth: 1,
        borderColor: 'grey',
        color:'grey',
        shadowColor: 'rgba(0,0,0, .4)', // IOS
        shadowOffset: { height: 1, width: 1 }, // IOS
        shadowOpacity: 1, // IOS
        shadowRadius: 1, //IOS
        elevation: 2, // Android
    },
    buttonStyle1:{
        height: 50,
        marginTop:10,
        paddingTop:15,
        paddingBottom:15,
        paddingLeft:30,
        paddingRight:30,
        marginLeft:10,
        backgroundColor:'green',
        borderRadius:10,
        borderWidth: 1,
        borderColor: 'green',
        color:'black',
        shadowColor: 'rgba(0,0,0, .4)', // IOS
        shadowOffset: { height: 1, width: 1 }, // IOS
        shadowOpacity: 1, // IOS
        shadowRadius: 1, //IOS
        elevation: 2, // Android
    },
    buttonStyle2:{
        height: 50,
        marginTop:10,
        paddingTop:12,
        paddingBottom:12,
        paddingLeft:30,
        paddingRight:30,
        marginLeft:10,
        backgroundColor:'red',
        borderRadius:10,
        borderWidth: 1,
        borderColor: 'red',
        color:'black',
        shadowColor: 'rgba(0,0,0, .4)', // IOS
        shadowOffset: { height: 1, width: 1 }, // IOS
        shadowOpacity: 1, // IOS
        shadowRadius: 1, //IOS
        elevation: 2, // Android
    }
})