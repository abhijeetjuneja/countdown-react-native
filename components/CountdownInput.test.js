import React from 'react';
import CountdownInput from './CountdownInput';

import renderer from 'react-test-renderer';
import { TouchableOpacity } from 'react-native'

//Snapshot tests
test('countdowninput renders correctly with countdown-0 and start-false', () => {
    const tree = renderer.create(<CountdownInput countdown={0} start={false}/>).toJSON();
    expect(tree).toMatchSnapshot();
});


test('countdowninput renders correctly with countdown-1 and start-false', () => {
    const tree = renderer.create(<CountdownInput countdown={0} start={false}/>).toJSON();
    expect(tree).toMatchSnapshot();
});


test('countdowninput renders correctly with countdown-1 and start-true', () => {
    const tree = renderer.create(<CountdownInput countdown={0} start={false}/>).toJSON();
    expect(tree).toMatchSnapshot();
});