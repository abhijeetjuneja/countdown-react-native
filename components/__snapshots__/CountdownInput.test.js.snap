// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`countdowninput renders correctly with countdown-0 and start-false 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginTop": 50,
    }
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "#0066FF",
        "borderColor": "#0066FF",
        "borderRadius": 10,
        "borderWidth": 2,
        "color": "white",
        "fontSize": 30,
        "fontWeight": "700",
        "paddingBottom": 15,
        "paddingLeft": 50,
        "paddingRight": 50,
        "paddingTop": 15,
        "width": "100%",
      }
    }
  >
    Countdown
  </Text>
  <br />
  <br />
  <Text>
    <TextInput
      allowFontScaling={true}
      disabled={false}
      onChange={[Function]}
      placeholder="Enter minutes"
      rejectResponderTermination={true}
      style={
        Object {
          "alignSelf": "center",
          "borderColor": "black",
          "borderRadius": 10,
          "borderWidth": 2,
          "fontSize": 16,
          "height": 50,
          "paddingLeft": 12,
          "width": "150px",
        }
      }
      underlineColorAndroid="transparent"
      value={0}
    />
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#C0C0C0",
          "borderColor": "grey",
          "borderRadius": 10,
          "borderWidth": 1,
          "color": "grey",
          "elevation": 2,
          "height": 50,
          "marginLeft": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingLeft": 30,
          "paddingRight": 30,
          "paddingTop": 15,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "grey",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        START
      </Text>
    </View>
  </Text>
</View>
`;

exports[`countdowninput renders correctly with countdown-1 and start-false 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginTop": 50,
    }
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "#0066FF",
        "borderColor": "#0066FF",
        "borderRadius": 10,
        "borderWidth": 2,
        "color": "white",
        "fontSize": 30,
        "fontWeight": "700",
        "paddingBottom": 15,
        "paddingLeft": 50,
        "paddingRight": 50,
        "paddingTop": 15,
        "width": "100%",
      }
    }
  >
    Countdown
  </Text>
  <br />
  <br />
  <Text>
    <TextInput
      allowFontScaling={true}
      disabled={false}
      onChange={[Function]}
      placeholder="Enter minutes"
      rejectResponderTermination={true}
      style={
        Object {
          "alignSelf": "center",
          "borderColor": "black",
          "borderRadius": 10,
          "borderWidth": 2,
          "fontSize": 16,
          "height": 50,
          "paddingLeft": 12,
          "width": "150px",
        }
      }
      underlineColorAndroid="transparent"
      value={0}
    />
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#C0C0C0",
          "borderColor": "grey",
          "borderRadius": 10,
          "borderWidth": 1,
          "color": "grey",
          "elevation": 2,
          "height": 50,
          "marginLeft": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingLeft": 30,
          "paddingRight": 30,
          "paddingTop": 15,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "grey",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        START
      </Text>
    </View>
  </Text>
</View>
`;

exports[`countdowninput renders correctly with countdown-1 and start-true 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginTop": 50,
    }
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "#0066FF",
        "borderColor": "#0066FF",
        "borderRadius": 10,
        "borderWidth": 2,
        "color": "white",
        "fontSize": 30,
        "fontWeight": "700",
        "paddingBottom": 15,
        "paddingLeft": 50,
        "paddingRight": 50,
        "paddingTop": 15,
        "width": "100%",
      }
    }
  >
    Countdown
  </Text>
  <br />
  <br />
  <Text>
    <TextInput
      allowFontScaling={true}
      disabled={false}
      onChange={[Function]}
      placeholder="Enter minutes"
      rejectResponderTermination={true}
      style={
        Object {
          "alignSelf": "center",
          "borderColor": "black",
          "borderRadius": 10,
          "borderWidth": 2,
          "fontSize": 16,
          "height": 50,
          "paddingLeft": 12,
          "width": "150px",
        }
      }
      underlineColorAndroid="transparent"
      value={0}
    />
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#C0C0C0",
          "borderColor": "grey",
          "borderRadius": 10,
          "borderWidth": 1,
          "color": "grey",
          "elevation": 2,
          "height": 50,
          "marginLeft": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingLeft": 30,
          "paddingRight": 30,
          "paddingTop": 15,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "grey",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        START
      </Text>
    </View>
  </Text>
</View>
`;
